src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -473,17 +473,15 @@ class MacroAssembler: public Assembler {
   void double_blt(FloatRegister Rs1, FloatRegister Rs2, Label &l, bool is_far = false, bool is_unordered = false);
   void double_bgt(FloatRegister Rs1, FloatRegister Rs2, Label &l, bool is_far = false, bool is_unordered = false);
 
-  void push_reg(RegSet regs, Register stack) { if (regs.bits()) { push_reg(regs.bits(), stack); } }
-  void pop_reg(RegSet regs, Register stack) { if (regs.bits()) { pop_reg(regs.bits(), stack); } }
   void push_reg(Register Rs);
   void pop_reg(Register Rd);
-  int  push_reg(unsigned int bitset, Register stack);
-  int  pop_reg(unsigned int bitset, Register stack);
-  void push_fp(FloatRegSet regs, Register stack) { if (regs.bits()) push_fp(regs.bits(), stack); }
-  void pop_fp(FloatRegSet regs, Register stack) { if (regs.bits()) pop_fp(regs.bits(), stack); }
+  void push_reg(RegSet regset, Register stack);
+  void pop_reg(RegSet regset, Register stack);
+  void push_fp(FloatRegSet regset, Register stack);
+  void pop_fp(FloatRegSet regset, Register stack);
 #ifdef COMPILER2
-  void push_vp(VectorRegSet regs, Register stack) { if (regs.bits()) push_vp(regs.bits(), stack); }
-  void pop_vp(VectorRegSet regs, Register stack) { if (regs.bits()) pop_vp(regs.bits(), stack); }
+  void push_vp(VectorRegSet regset, Register stack);
+  void pop_vp(VectorRegSet regset, Register stack);
```

Comment:
The function name push_vp / pop_vp look a bit strange to me. I think we should rename them to push_v / pop_v.

Reply:
Done, renamed to `push_v/pop_v`.